test/jdk/java/lang/reflect/code/TestErasure.java

```diff
@@ -117,13 +118,13 @@ static List<TypeAndErasure> typeVars() {
         List<TypeAndErasure> typeVars = new ArrayList<>();
         for (int dims = 1 ; dims <= 3 ; dims++) {
             for (TypeAndErasure t : references()) {
-                typeVars.add(new TypeAndErasure(JavaType.typeVarRef("X", JavaType.J_L_OBJECT, t.type), t.erasure));
+                typeVars.add(new TypeAndErasure(JavaType.typeVarRef("X", (ClassType)JavaType.J_L_OBJECT, t.type), t.erasure));
```

Comment:
I plan to sharpen the types of the various `JavaType` constants in a follow up PR where I'll also clean up the various type factories.